### PR TITLE
Backtracking

### LINKED ISSUE #1597: Backtracking using Bitmasking
I would like to contribute to this problem for GSSoC'20.

### PR DESCRIPTION
**Pull request by RiaJha02**
Fixes 
#1597
Backtracking Using Bitmasking
Backtracking is an algorithmic-technique for solving problems recursively by trying to build a solution incrementally, one piece at a time, removing those solutions that fail to satisfy the constraints of the problem at any point of time.
Here, we'll be taking the case of a backtracking based problem: N Queen problem.
The N Queen is the problem of placing N chess queens on an N×N chessboard so that no two queens attack each other.
We'll be implementing the solution using backtracking and bit masking in various programming languages.

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Languages Used:
- CPP
- JAVA
- Python

#### Files Added:
- Backtrack_using_bitmask.cpp
- Backtrack_using_bitmask.java
- Backtrack_using_bitmask.py
[Main.Documents.pdf](https://github.com/jainaman224/Algo_Ds_Notes/files/4363627/Main.Documents.pdf)

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by RiaJha02** [comment]
#1597

File: Backtracking_using_bitmask/Backtrack_using_bitmask.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#include <bits/stdc++.h>
+using namespace std;
+#define d (1 << n) - 1
+int cnt = 0;
+//This is an optimised approach than the normal backtracking approach.
+//Here we don’t need to write is Safe Position Function
+//which works in linear time instead we use bitsets which work in O(1)
+time.
```
Comment (asha15) [inline Backtracking_using_bitmask/Backtrack_using_bitmask.cpp:8]: make this as a multi line comment

**Comment by RiaJha02** [comment]
Done with the changes, Ma'am.

**Comment by asha15** [comment]
Good job!

please squash your commits we are good to merge

**Pull request merged**